server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestPutIndexTemplateAction.java

```diff
@@ -42,6 +42,8 @@
 
     private static final DeprecationLogger deprecationLogger = new DeprecationLogger(
             LogManager.getLogger(RestPutIndexTemplateAction.class));
+    public static final String TYPES_DEPRECATION_MESSAGE = "[types removal]" +
```

Comment:
Very small comment: we could add the sentence "The parameter will be removed in the next major version." for consistency with our other warnings.